crates/api_models/src/payment_methods.rs

```diff
@@ -72,7 +73,7 @@ pub struct PaymentMethodUpdate {
 pub struct CardDetail {
     /// Card Number
     #[schema(value_type = String,example = "4111111145551142")]
-    pub card_number: masking::Secret<String, pii::CardNumber>,
+    pub card_number: masking::Secret<String, CardNumberStrategy>,
```

Comment:
You could just use `CardNumber` as the type instead. Please make this change elsewhere too.

```suggestion
    pub card_number: CardNumber,
```